video/image_writer.c

```diff
@@ -430,6 +440,11 @@ static bool write_avif(struct image_writer_ctx *ctx, mp_image_t *image,
         goto free_data;
     }
 
+    if (!overwrite && mp_path_exists(filename)) {
+        MP_ERR(ctx, "File '%s' already exists\n", filename);
+        goto free_data;
+    }
+
     ret = avio_open(&avioctx, filename, AVIO_FLAG_WRITE);
```

Comment:
you can use `avio_open_dyn_buf` (or a custom avio context, but that's overkill) to gain control over file opening/writing

and then to deduplicate code you can refactor all `write_`* functions to directly take the FILE pointer

Reply:
That's actually good idea to refactor everything and save into memory and save to file with one common function. Whether it is worth the changes needed is another question.

Reply:
I agree it is probably not worth the changes, especially if it is not bringing clear value.

Reply:
Personally I'd do the `FILE*` change but not the write-to-memory part. (except for `write_avif` where this is a convenient way of working with the file pointer)